### PR TITLE
Security secure connections only.

### PR DESCRIPTION
**Pull request by movrsi**
open-in-mpv should only be allowing secure connections; perhaps the extension can have an option to change this setting, but by default secure connections should only be allowed to be passed to open-in-mpv.

### CODE REVIEW

File: open_in_mpv/open_in_mpv.py [hunk 1]
```diff
@@ -193,6 +193,10 @@ def real_main(log: TextIO) -> int:
         logger.exception('No URL was given')
         print(json.dumps(dict(message='Missing URL!')))
         return 1
+    if 'https' not in url:
```
Comment (Tatsh) [inline open_in_mpv/open_in_mpv.py:196]: There should be a setting.

File: open_in_mpv/open_in_mpv.py [hunk 2]
```diff
@@ -193,6 +193,10 @@ def real_main(log: TextIO) -> int:
...
+        logger.exception('The url appears to not have a secure connection.')
+        print(json.dumps(dict(message='Security Exception!')))
```
Comment (Tatsh) [inline open_in_mpv/open_in_mpv.py:198]: Just put the message in the JSON and remove the logger call. Also, no shouting.

**Review by Tatsh — CHANGES_REQUESTED** [review]

**Comment by Tatsh** [comment]
> QA failing here is odd;
> 
> ```python
> 'x' not in a
> ```
> 
> is valid Python.

It is complaining because line 192 is written incorrectly. There should only be `()` around the assignment and not the `is None` part.

File: open_in_mpv/open_in_mpv.py (hunk 1 above)
Comment (movrsi) [reply open_in_mpv/open_in_mpv.py:196]: yeah the idea is to integrate this into the extension so the user can decide if they wish to load non secure urls. but by default open-in-mpv should disallow, have to have a json response from the extension telling open-in-mpv to disregard this check.

File: open_in_mpv/open_in_mpv.py (hunk 2 above)
Comment (movrsi) [reply open_in_mpv/open_in_mpv.py:198]: Well shouting occurs if the URL; but no problem ill fix that.

**Comment by movrsi** [comment]
> > QA failing here is odd;
> > ```python
> > 'x' not in a
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > is valid Python.
> 
> It is complaining because line 192 is written incorrectly. There should only be `()` around the assignment and not the `is None` part.

Heh, i thought that was fixed before the cleanup was merged. regardless corrected it.

File: open_in_mpv/open_in_mpv.py [hunk 3]
```diff
@@ -189,10 +189,13 @@ def real_main(log: TextIO) -> int:
         response(dict(version=VERSION, logPath=log.name, socketPath=MPV_SOCKET))
         log.close()
         return 0
-    if (url := message.get('url', None) is None):
+    if (url := message.get('url', None)) is None:
         logger.exception('No URL was given')
         print(json.dumps(dict(message='Missing URL!')))
         return 1
+    if 'https' not in url:
+        print(json.dumps(dict(message='Security blocked unsecure connection.')))
```
Comment (Tatsh) [inline open_in_mpv/open_in_mpv.py:197]: Make it say: *Insecure URLs (non-HTTPS) are blocked by default.*

Comment (movrsi) [reply open_in_mpv/open_in_mpv.py]: Done.

**Pull request merged**